test/arithmetic/multiplier_test.dart

```diff
@@ -0,0 +1,221 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// multiplier_test.dart
+// Test Multiplier and MultiplerAccumulate:  CompressionTree implementations
+//
+// 2024 August 7
+// Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'dart:math';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+import 'package:test/test.dart';
+
+// Inner test of a multipy accumulate unit
+void checkMultiplyAccumulate(
+    MultiplyAccumulate mod, BigInt bA, BigInt bB, BigInt bC) {
+  final golden = bA * bB + bC;
+  // ignore: invalid_use_of_protected_member
+  mod.a.put(bA);
+  // ignore: invalid_use_of_protected_member
+  mod.b.put(bB);
+  // ignore: invalid_use_of_protected_member
+  mod.c.put(bC);
+  // print('$bA, $bB, $bC');
+
+  final result = mod.signed
+      ? mod.accumulate.value.toBigInt().toSigned(mod.accumulate.width)
+      : mod.accumulate.value.toBigInt().toUnsigned(mod.accumulate.width);
+  expect(result, equals(golden));
+}
+
+// Random testing of a mutiplier or multiplier/accumulate unit
+void testMultiplyAccumulateRandom(int width, int iterations,
+    MultiplyAccumulate Function(Logic a, Logic b, Logic c) fn) {
+  final a = Logic(name: 'a', width: width);
+  final b = Logic(name: 'b', width: width);
+  final c = Logic(name: 'c', width: width * 2);
+  final mod = fn(a, b, c);
+  test('random_${mod.name}_S${mod.signed}_W${width}_I$iterations', () async {
+    final multiplyOnly = mod is MultiplyOnly;
+    await mod.build();
+    final signed = mod.signed;
+    for (var i = 0; i < iterations; i++) {
+      final bA = signed
+          ? Random().nextLogicValue(width: width).toBigInt().toSigned(width)
```

Comment:
use a single `Random` object created in one place (maybe per-test?) with a fixed seed provided so that tests are reproducible

Reply:
done.